pkg/apis/provisioning/v0alpha1/types.go

```diff
@@ -91,6 +96,16 @@ type RepositorySpec struct {
 	GitHub *GitHubRepositoryConfig `json:"github,omitempty"`
 }
 
+// The status of a Repository.
+// This is expected never to be created by a kubectl call or similar, and is expected to rarely (if ever) be edited manually.
+// As such, it is also a little less well structured than the spec, such as conditional-but-ever-present fields.
+type RepositoryStatus struct {
+	// The Git commit we're currently synced to.
+	// A non-empty value only matters if we use a git storage backend.
+	// Useful for no-clone Git clients and if cloning Git clients ever lose their clones.
```

Comment:
should this then be a pointer to a string? It would make it clearer when it does not make sense.

Reply:
It could be. I just didn't want to write a `String` method or similar workarounds to be able to reasonably inspect it in logs when testing 😄

Do you have any strong preference?

Reply:
omitempty behaves the same in the OpenAPI spec and is much easier to work with!